src/libraries/System.Private.CoreLib/src/System/Reflection/Assembly.cs

```diff
@@ -217,13 +217,31 @@ public override string ToString()
             return type.Module?.Assembly;
         }
 
-        // internal test hook
-        private static bool s_forceNullEntryPoint;
+        private static Assembly? s_overriddenEntryAssembly;
+
+        /// <summary>
+        /// Sets the application's entry assembly to the provided assembly object
+        /// as argument.
+        /// </summary>
+        /// <param name="assembly">
+        /// Assembly object that represents the application's new entry assembly.
+        /// </param>
+        /// <remarks>
+        /// It is important to mention that the assembly passed to this function
+        /// has to be a RuntimeAssembly type. Otherwise, an exception will be thrown.
```

Comment:
`RuntimeAssembly` type is internal implementation detail of the runtime. We do not document what it is.

This should say "a runtime Assembly object" or "runtime defined Assembly object".